postgres_globals.py

```diff
@@ -5,205 +5,213 @@
-                      sum(autovacuum_count) AS sum_autovacuum_count,
-                      sum(analyze_count) AS sum_analyze_count,
-                      sum(autoanalyze_count) AS sum_autoanalyze_count
-               FROM pg_stat_user_tables"""
-    now = str(int(time.time()))
-
-    for line in connect_and_execute(query, database):
         for key, value in line.items():
             if value is not None:
-                print(template.format(key, value, now))
+                print(template.format(datname, key, value))
+
+    #
+    # Statistics by database system tables
+    #
+    for database in user_databases:
+        template = (
+            args.prefix + "db." + database + ".stat_sys_tables.{0} {1} " +
+            str(int(time()))
+        )
+        query = """SELECT sum(seq_scan) AS sum_seq_scan,
+                        sum(seq_tup_read) AS sum_seq_tup_read,
+                        sum(idx_scan) AS sum_idx_scan,
+                        sum(idx_tup_fetch) AS sum_idx_tup_fetch,
+                        sum(n_tup_ins) AS sumn_tup_ins,
+                        sum(n_tup_upd) AS sum_n_tup_upd,
+                        sum(n_tup_del) AS sum_n_tup_del,
+                        sum(n_tup_hot_upd) AS sum_n_tup_hot_upd,
+                        sum(n_live_tup) AS sum_n_live_tup,
+                        sum(n_dead_tup) AS sum_n_dead_tup,
+                        sum(vacuum_count) AS sum_vacuum_count,
+                        sum(autovacuum_count) AS sum_autovacuum_count,
+                        sum(analyze_count) AS sum_analyze_count,
+                        sum(autoanalyze_count) AS sum_autoanalyze_count
+                FROM pg_stat_sys_tables"""
+
+        for line in connect_and_execute(query, database):
+            for key, value in line.items():
+                print(template.format(key, value))
+
+    #
+    # Statistics by database user tables
+    #
+    for database in user_databases:
+        template = (
+            args.prefix + "db." + database + ".stat_user_tables.{0} {1} " +
+            str(int(time()))
+        )
+        query = """SELECT sum(seq_scan) AS sum_seq_scan,
+                        sum(seq_tup_read) AS sum_seq_tup_read,
+                        sum(idx_scan) AS sum_idx_scan,
+                        sum(idx_tup_fetch) AS sum_idx_tup_fetch,
+                        sum(n_tup_ins) AS sumn_tup_ins,
+                        sum(n_tup_upd) AS sum_n_tup_upd,
+                        sum(n_tup_del) AS sum_n_tup_del,
+                        sum(n_tup_hot_upd) AS sum_n_tup_hot_upd,
+                        sum(n_live_tup) AS sum_n_live_tup,
+                        sum(n_dead_tup) AS sum_n_dead_tup,
+                        sum(vacuum_count) AS sum_vacuum_count,
+                        sum(autovacuum_count) AS sum_autovacuum_count,
+                        sum(analyze_count) AS sum_analyze_count,
+                        sum(autoanalyze_count) AS sum_autoanalyze_count
```

Comment:
It is completely the same selection right? Can't we somehow store that or maybe wrap it in a function with a parameter for the table.